server/settings_test.go

```diff
@@ -618,20 +622,65 @@ groups:
 
 					res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
 						Body: server.ChangeSettingsBody{
-							AlertManagerURL:   url,
+							AlertManagerURL:   alertManagerURL,
 							AlertManagerRules: rules,
 						},
 						Context: pmmapitests.Context,
 					})
 					require.NoError(t, err)
-					assert.Equal(t, url, res.Payload.Settings.AlertManagerURL)
+					assert.Equal(t, alertManagerURL, res.Payload.Settings.AlertManagerURL)
 					assert.Equal(t, rules, res.Payload.Settings.AlertManagerRules)
 
 					gets, err := serverClient.Default.Server.GetSettings(nil)
 					require.NoError(t, err)
-					assert.Equal(t, url, gets.Payload.Settings.AlertManagerURL)
+					assert.Equal(t, alertManagerURL, gets.Payload.Settings.AlertManagerURL)
 					assert.Equal(t, rules, gets.Payload.Settings.AlertManagerRules)
 
+					t.Run("PrometheusReloaded", func(t *testing.T) {
+						prometheusBaseURL := pmmapitests.BaseURL.ResolveReference(&url.URL{
+							Path: "prometheus/",
+						})
+						client, err := prometheusApi.NewClient(prometheusApi.Config{
+							Address: prometheusBaseURL.String(),
+						})
+						require.NoError(t, err)
+
+						expected := prometheusApiV1.RuleGroup{
+							Rules: prometheusApiV1.Rules{
+								prometheusApiV1.AlertingRule{
+									Alerts:      []*prometheusApiV1.Alert{},
+									Annotations: model.LabelSet{"summary": "High request latency"},
+									Duration:    600,
+									Health:      "unknown",
+									Labels:      model.LabelSet{"severity": "page"},
+									Name:        "HighRequestLatency",
+									Query:       `job:request_latency_seconds:mean5m{job="myjob"} > 0.5`,
+								},
+							},
+							File:     "/srv/prometheus/rules/pmm.rules.yml",
+							Interval: 60,
+							Name:     "example",
+						}
+
+						// Prometheus needs some time to reload configs.
+						ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+						defer cancel()
+
+						api := prometheusApiV1.NewAPI(client)
+						for {
+							// Get rules using API client.
```

Comment:
My comments were for explaining what should be do there. It doesn't look particularly useful in real code.
![image](https://user-images.githubusercontent.com/11512/83889542-f1424e00-a753-11ea-98fb-996863d52d3c.png)